src/stactools/cli/commands/validate.py

```diff
@@ -15,25 +17,42 @@ def create_validate_command(cli):
     @click.option("--recurse/--no-recurse",
                   default=True,
                   help=("If false, do not validate any children "
-                        "(only useful for Catalogs and Collections"))
+                        "(only useful for Catalogs and Collections)"))
     @click.option("--links/--no-links",
                   default=True,
                   help=("If false, do not check any of the objects's links."))
     @click.option(
         "--assets/--no-assets",
         default=True,
         help=("If false, do not check any of the collection's/item's assets."))
-    def validate_command(href, recurse, links, assets):
+    @click.option("--stac-vrt",
+                  is_flag=True,
+                  default=False,
+                  help=("If true, verify that the required fields exist "
+                        "for use by the stac-vrt tool. (default: false)"))
+    def validate_command(href, recurse, links, assets, stac_vrt):
         """Validates a STAC object.
 
         Prints any validation errors to stdout.
         """
         object = pystac.read_file(href)
 
-        if isinstance(object, Item):
-            errors = validate(object, None, False, links, assets)
-        else:
-            errors = validate(object, object, recurse, links, assets)
+        validators = [None]  # Start with the existing default
+        if stac_vrt:
+            validators.append(
+                RequiredFieldsValidator({
+                    STACObjectType.ITEM:
+                    ["proj:bbox", "proj:epsg", "proj:shape", "proj:transform"]
+                }))
+
+        errors = []
+        for validator in validators:
+            if isinstance(object, Item):
+                errors += root_validate(validator, object, None, False, links,
+                                        assets)
+            else:
+                errors += root_validate(validator, object, object, recurse,
+                                        links, assets)
```

Comment:
```suggestion
        validators.insert(0, None)
        for validator in validators:
            pystac.validators.set_validator(validator)
            if isinstance(object, Item):
                errors += validate(object, None, False, links, assets)
            else:
                errors += validate(object, object, recurse, links, assets)
        pystac.validators.set_validator(None)
```

I don't know if we need the context manager, since the "default" validator activated by setting it to `None`.

Reply:
I was using the context manager for two reasons. One was to ensure that the validator would be set to the pre-existing setting, even if an uncaught exception stopped our command from completing. With the setting being global, any other use of `object.validate()` will use whatever validator threw the exception. [I think that at the moment this won't happen because we catch all exceptions]
The other was to allow users to change the main validator. That isn't really an issue at the moment, but if people are using `stactools` programmatically it might cause grief, since running our `validate` will change the behaviour of future `object.validate()`s.
Another option would be to add a `validate(self, validator)` to `STACObject` in `pystac` which would remove the need for this.

Reply:
Ah gotcha, ok the context manager makes sense then.

>  Another option would be to add a validate(self, validator) to STACObject in pystac which would remove the need for this.

I agree with this, much prefer declarative instead of global-ish magic :-).